### ISSUE TITLE
CI unstable due to downloading golang from dl.google.com

### ISSUE THREAD

**Issue by twelho** [weaveworks/wksctl#217]
Multiple tasks on CircleCI build the Docker image at `test/container/images/centos7/Dockerfile`, which downloads golang using `curl -fsSLo /tmp/gotar.tgz "https://dl.google.com/go/go$GO_VERSION.linux-$ARCH.tar.gz"`.

That operation is unrealiable for some reason, often ending in a transient problem or the transfer closing abruptly, failing CI. Retry options to `curl` don't seem to improve the situation. Building the image locally I don't hit these issues.

This affects https://github.com/weaveworks/wksctl/pull/216 and https://github.com/weaveworks/wksctl/pull/214 among others.

**Comment by bboreham** [comment]
Could you give some specific CI links to failures?
I clicked through four on https://circleci.com/gh/weaveworks/wksctl/tree/pull%2F214?page=1 but didn't find the problem you mentioned.

**Comment by twelho** [comment]
Sure, here's one failed run in `build` [#5502](https://app.circleci.com/pipelines/github/weaveworks/wksctl/893/workflows/e0e18103-b2a2-4215-b7bf-c81f52d9a4a5/jobs/5502) and one in `container-tests` [#5503](https://app.circleci.com/pipelines/github/weaveworks/wksctl/894/workflows/1eab8640-d907-405c-be27-e8fdb034dc8b/jobs/5503) where I explicitly tried giving `curl` some retry parameters without success. I've been hitting the rerun button for both #216 and #214 to get the CI to pass, so apparently the latter doesn't have those errors on record anymore.

**Comment by bboreham** [comment]
Can you figure out why it's downloading Go at that particular point in the build?
It seems to be building `weaveworks/wksctl-centos7` but I can't find any references to that in my source tree.

**Comment by twelho** [comment]
I'll take a look. Based off a quick look the image might be unused, at least during `build` there doesn't seem to be a reason to build it.

**Comment by twelho** [comment]
`weaveworks/wksctl-centos7` is not needed for `build`, that was a error introduced in #216 has now been fixed. It is though explicitly marked as a dependency for `container-tests: test/container/images/centos7/.uptodate`, but as far as I see those tests only reference `quay.io/wksctl/centos7` from `test/container/images/names.go`.

Is `test/container/images/centos7/Dockerfile` possibly the base for `quay.io/wksctl/centos7`? [quay.io shows the latter hasn't been updated for 10 months](https://quay.io/repository/wksctl/centos7?tab=tags).

**Comment by twelho** [comment]
Upon further inspection, `weaveworks/wksctl-centos7` is completely unused (CI passes if it's removed): https://github.com/weaveworks/wksctl/pull/216/commits/738165d4393ae824a80bc300d3e31188e8f6b800

It also seems that `quay.io/wksctl/centos7` is very similar to that build image, from what I can tell it lacks golang and has a different `SRC_PATH` (ends in `wks` instead of `wksctl`), those seem to be the only differences.

**Comment by bboreham** [comment]
This view on Quay does correspond very well to that Dockerfile: https://quay.io/repository/wksctl/centos7/manifest/sha256:2df8def31eae70771dd298d860f221cfaa83b16f4add2d9644c955726ef30c59

I'm still struggling to see how the comment `# Install golang so we can run our tests on macs` applies - none of the tests using that image appear to build Go programs inside it.

Other tests use `quay.io/footloose/centos7:0.6.0`

**Comment by twelho** [comment]
Right, the go libraries and binaries are present both in the built image and `quay.io/wksctl/centos7`, but they're not even added to `$PATH`. Is there a build system present for `quay.io/wksctl/centos7` somewhere?

**Comment by twelho** [comment]
It appears that this Dockerfile originates from the closed parts before this project was open sourced, since it was added in the initial commit: https://github.com/weaveworks/wksctl/commits/master/test/container/images/centos7/Dockerfile

Since nothing really uses it anymore and the workflow has changed since then (no manual testing on macs), I assume it can be removed?

**Comment by bboreham** [comment]
I think it's fine to remove it; we can always find the old code from git history and put it back if this is a mistake.

**Comment by twelho** [comment]
:+1: #216 already tracks a lot of changes, so the removal will happen in #218.
